fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/directory/MockedFDBDirectoryWrapper.java

```diff
@@ -31,8 +31,11 @@
  * A Testing-focused {@link FDBDirectoryWrapper} that allows a mocked-FDBDirectory to be injected into the system.
  */
 public class MockedFDBDirectoryWrapper extends FDBDirectoryWrapper {
-    MockedFDBDirectoryWrapper(final IndexMaintainerState state, final Tuple key, final int mergeDirectoryCount, final AgilityContext agilityContext, final int blockCacheMaximumSize) {
+    MockedFDBDirectoryWrapper(final IndexMaintainerState state, final Tuple key, final int mergeDirectoryCount, final AgilityContext agilityContext, final int blockCacheMaximumSize, final InjectedFailureRepository injectedFailures) {
         super(state, key, mergeDirectoryCount, agilityContext, blockCacheMaximumSize);
+        // Set the injectedFailures at the end of the constructor since createDirectory() is called from the constructor
+        // and we can't pass the injected failures to it yet.
```

Comment:
I couldn’t see how to achieve that. For the two cases:
1.
```
MockedFDBDirectoryWrapper -> super() -> createFDBDirectory()
```
The overridden `createFDBDirectory` cannot access the `injectedFailures` from the `MockedFDBDirectoryWrapper` since we're still in the context of the constructor and local state has not been set yet.
2.
```
MockedLuceneIndexMaintainer -> super() -> createDirectoryManager()
```
The overridden `createDirectoryManager` does not have access to the `injectedFailures` from the constructor since local state has not been set yet, so it cannot pass that along to the `getManager()`.